app/src/main/java/nl/quintor/studybits/studybitswallet/IndyClient.java

```diff
@@ -92,17 +92,18 @@ public MessageEnvelope fulfillExchangePosition(ExchangePosition exchangePosition
     }
 
     @NonNull
-    public University connect(String endpoint, AgentClient agentClient, ConnectionRequest connectionRequest) throws InterruptedException, ExecutionException, IndyException, IOException {
+    public University connect(String endpoint, String uniName, String username, String uniVerinymDid, AgentClient agentClient) throws InterruptedException, ExecutionException, IndyException, IOException {
+        ConnectionRequest connectionRequest = studentWallet.createConnectionRequest(uniVerinymDid).get();
 
-        ConnectionResponse connectionResponse = studentWallet.acceptConnectionRequest(connectionRequest).get();
+        MessageEnvelope<ConnectionRequest> connectionResponseEnvelope = studentCodec.encryptMessage(connectionRequest, IndyMessageTypes.CONNECTION_REQUEST).get();
 
-        MessageEnvelope<ConnectionResponse> connectionResponseEnvelope = studentCodec.encryptMessage(connectionResponse, IndyMessageTypes.CONNECTION_RESPONSE).get();
+        MessageEnvelope<ConnectionResponse> connectionResponseMessageEnvelope = agentClient.login(username, connectionResponseEnvelope);
```

Comment:
Shouldn't this need a password aswell? It won't be able to login to the dummy student 'Lisa' without a password.